libs/qCC_db/src/ccScalarField.cpp

```diff
@@ -205,7 +205,7 @@ void ccScalarField::computeMinAndMax()
 
 						if (ValidValue(val))
 						{
-							unsigned bin = static_cast<unsigned>(floor(static_cast<double>(val - m_displayRange.min())*step));
+							unsigned bin = static_cast<unsigned>((val - m_displayRange.min())*step);
```

Comment:
why removing the call to 'floor'? This changes the result... and I believe 'floor' was the right behavior 👼

Reply:
I do not believe this changes the result. According to [here](https://cplusplus.com/doc/tutorial/typecasting/), a float to int cast truncates the value, i.e. removes the decimal part. The expression is never negative, so that produces the same result as flooring it.

As for why to remove it, the promotion to double and the call to floor cause a substantial overhead. Removing both roughly doubles execution speed of this function.

Can you help me understand why you believe the result changes?

Reply:
I think floor and cast are not equivalent when the value is negative. is this value could be negative before being converted to unsigned?

Reply:
Well, that was bugging me as well, but I don't see how it would have worked either with floor if the value could be smaller than m_displayRange.min() 🤔 

This code is quite old, I don't remember much about it.

Reply:
m_displayRange.min() is set to the scalar field's minimum value at the start of the function [here](https://github.com/CloudCompare/CloudCompare/blob/5aec741ad191bb88da267f6fd8157f99142c4807/libs/qCC_db/src/ccScalarField.cpp#L167), so that expression can't be negative. There can't be any values in the scalar field lower than its minimum value.